File: refactoring_test/rscore/tests/application/AddPrefixToPrivateFieldsTest.scala
package rscore.tests.application
import org.junit.Test
import rscore.dsl.entity.collection.By
import rscore.dsl.entity.collection.With

class AddPrefixToPrivateFieldsTest extends ApplicationTest{
	@Test
	def 純粋privateなフィールドの先頭にプレフィクスがついていなければつける(): Unit = {
		val testName = "AddPrefixToPrivateFieldsTest"
			prepareTest(testName)
		
		getCurrentProject().pkg(this.testGroupIdentifier).classes().foreach(c => {
			c.fields.select(By modifier "private").
			select(By modifier (With out "static")).foreach(f => {
				val name = f.name
				if(name.charAt(0) != '_'){
					f.rename("_" + name)
				}
			})
		})
		
		doAssert(testName)
	}
	

}

File: test_resources_output/application/AddPrefixToPrivateFieldsTest.java
package application;

public class AddPrefixToPrivateFieldsTest {
	public int a;
	private int _b;
	protected int c;
	int d;
	
	void m(){
		System.out.println(a + _b + c + d);
	}

}

class Foo{
	public int a;
	private int _b;
	protected int c;
	int d;
	
	void m(){
		System.out.println(a + _b + c + d);
	}
}

class Bar{
	public int a;
	private int _b;
	protected int c;
	int d;
	
	void m(){
		System.out.println(a + _b + c + d);
	}
}

class Hoge{
	public static int a;
	private static int b;
	protected static int c;
	static int d;
	void m(){
		System.out.println(a + b + c + d);
	}
}
